crates/holochain/src/conductor/interface.rs

```diff
@@ -1,145 +1,36 @@
-use crate::conductor::api::*;
-use error::InterfaceResult;
-use futures::{
-    channel::mpsc::{channel, Receiver, Sender},
-    future::{BoxFuture, FutureExt},
-    sink::{Sink, SinkExt},
-    stream::{Stream, StreamExt},
-};
-use holochain_serialized_bytes::{SerializedBytes, SerializedBytesError};
+use crate::{conductor::api::*, core::signal::Signal};
+use error::{InterfaceError, InterfaceResult};
 use serde::{Deserialize, Serialize};
-use std::convert::{TryFrom, TryInto};
+use std::convert::TryInto;
+use tokio::sync::broadcast;
 
 pub mod error;
 pub mod websocket;
 
-/// Allows the conductor or cell to forward signals to connected clients
-pub struct ConductorSideSignalSender<Sig>
-where
-    Sig: 'static + Send + TryInto<SerializedBytes, Error = SerializedBytesError>,
-{
-    sender: Sender<SerializedBytes>,
-    phantom: std::marker::PhantomData<Sig>,
-}
+#[derive(Clone, Debug)]
+pub struct SignalBroadcaster(Vec<broadcast::Sender<Signal>>);
 
-impl<Sig> ConductorSideSignalSender<Sig>
-where
-    Sig: 'static + Send + TryInto<SerializedBytes, Error = SerializedBytesError>,
-{
+impl SignalBroadcaster {
     /// send the signal to the connected client
-    pub async fn send(&mut self, data: Sig) -> InterfaceResult<()> {
-        let data: SerializedBytes = data.try_into()?;
-        self.sender.send(data).await?;
+    pub async fn send(&mut self, sig: Signal) -> InterfaceResult<()> {
+        self.0
+            .iter_mut()
+            .map(|tx| tx.send(sig.clone()))
+            .collect::<Result<Vec<_>, broadcast::SendError<Signal>>>()
+            .map_err(InterfaceError::SignalSend)?;
```

Comment:
This could be just one sender to all the receivers maybe?

Reply:
see above